packages/ckeditor5-alignment/src/alignmentui.ts

```diff
@@ -122,50 +191,62 @@ export default class AlignmentUI extends Plugin {
 	}
 
 	/**
-	 * Helper method for initializing the button and linking it with an appropriate command.
+	 * Creates a menu for all alignment options to use either in menu bar.
 	 *
-	 * @param option The name of the alignment option for which the button is added.
+	 * @param options Normalized alignment options from config.
 	 */
-	private _addButton( option: SupportedOption ): void {
+	private _addMenuBarMenu( options: Array<AlignmentFormat> ): void {
 		const editor = this.editor;
 
-		editor.ui.componentFactory.add( `alignment:${ option }`, locale => this._createButton( locale, option ) );
-	}
+		editor.ui.componentFactory.add( 'menuBar:alignment', locale => {
+			const command: AlignmentCommand = editor.commands.get( 'alignment' )!;
+			const t = locale.t;
+			const menuView = new MenuBarMenuView( locale );
+			const listView = new MenuBarMenuListView( locale );
 
-	/**
-	 * Helper method for creating the button view element.
-	 *
-	 * @param locale Editor locale.
-	 * @param option The name of the alignment option for which the button is added.
-	 * @param buttonAttrs Optional parameters passed to button view instance.
-	 */
-	private _createButton(
-		locale: Locale,
-		option: SupportedOption,
-		buttonAttrs: Partial<Button> = {}
-	): ButtonView {
-		const editor = this.editor;
-		const command: AlignmentCommand = editor.commands.get( 'alignment' )!;
-		const buttonView = new ButtonView( locale );
+			menuView.bind( 'isEnabled' ).to( command );
 
-		buttonView.set( {
-			label: this.localizedOptionTitles[ option ],
-			icon: iconsMap.get( option ),
-			tooltip: true,
-			isToggleable: true,
-			...buttonAttrs
-		} );
+			listView.set( {
+				ariaLabel: t( 'Text alignment' ),
+				role: 'menu'
+			} );
 
-		// Bind button model to command.
-		buttonView.bind( 'isEnabled' ).to( command );
-		buttonView.bind( 'isOn' ).to( command, 'value', value => value === option );
+			menuView.buttonView.set( {
+				label: t( 'Text alignment' )
+			} );
 
-		// Execute command.
-		this.listenTo( buttonView, 'execute', () => {
-			editor.execute( 'alignment', { value: option } );
-			editor.editing.view.focus();
-		} );
+			for ( const option of options ) {
+				const listItemView = new MenuBarMenuListItemView( locale, menuView );
+				const buttonView = new MenuBarMenuListItemButtonView( locale );
 
-		return buttonView;
+				buttonView.extendTemplate( {
+					attributes: {
+						'aria-checked': buttonView.bindTemplate.to( 'isOn' )
```

Comment:
I wonder if this could be just default behavior of a button. But then `isOn` would need to be able to take `undefined` state.